android/src/main/java/org/mozilla/experiments/nimbus/Variables.kt

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
This file defines an interface `Variables`, which gives the user a generic view of the config, with some helper methods to make it easy to write Android code with. It also provides two implementations: one a thin wrapper for a JSON object, and the second which makes calls across the FFI.

For nested configurations, the JSON implementation is used for both.